portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntryTag.testcase

```diff
@@ -522,6 +522,9 @@ definition {
 			Questions.search(
 				searchKey = "tag1",
 				searchTag = "true");
+
+			// Pausing 1 second for giving the time to perform the search LPD-27732
+			Pause(locator1 = 1000);
```

Comment:
I think Brian will not approve this, if I'm not mistaken, Pause is no longer allowed. Is that correct @austinchiang ? What would be the correct approach now?

Reply:
It was suggested by CI team some days ago in slack: https://liferay.slack.com/archives/C19PWQ74J/p1716575155515559?thread_ts=1716542537.448449&cid=C19PWQ74J 
There are cases that we may not have other option. But I'm open to anny other idea about how to wait for a search done with graphql in poshi.

Reply:
The last time I checked you will need to justify using the pause in the comment. It's discouraged to use pauses but there are still exceptions

Reply:
And for this particular case (sometimes search is performed and the tag2 is not present, but sometimes search hasn't finished and tag2 is still present and test fails), could you suggest a different fix instead of using the Pause? I have no other ideas...

Reply:
Hmm in this case I can think of a few options:
1. WaitForPageLoad() - this waits for the entire page to be loaded. No need to specify a locator
2. WaitForElementPresent() - here you would specify an element/locator that only appears after the search is performed, maybe the search result text or something else. The assumption is that if that element is present on the page then the search has finished and everything else is loaded properly

Reply:
I think `WaitForPageLoad` would not work in this case. Questions is build in react(all in client side), so the page is loaded and a search does not load a new page but changes the markup (or that is what makes sense to me). In this case I'm going to try with `WaitForElementNotPresent`, because the idea of the test is that there are `tag1`, `tag2` and `tag3`, and when search for `tag1`, then `tag2` and `tag3` must be hidden. 

```
		task ("When a tag is searched") {
			Questions.openToTagsTab(
				layoutName = "questions-page",
				siteURLKey = "guest");

			Questions.search(
				searchKey = "tag1",
				searchTag = "true");

			WaitForElementNotPresent(
				key_tagName = "tag2",
				locator1 = "Questions#TAGS_LIST_TAB_NAMES");
		}

		task ("Then only the matching tag is displayed") {
			Questions.viewTags(tagName = "tag1");

			Questions.viewNoTags(tagsList = "tag2,tag3");

			Click(
				ariaLabel = "Clear",
				locator1 = "Button#ANY_WITH_ARIA_LABEL");

			for (var tagName : list "tag1,tag2,tag3") {
				Questions.viewTags(tagName = ${tagName});
			}
```